crates/bevy_ui/src/update.rs

```diff
@@ -37,35 +37,46 @@ fn update_clipping(
     entity: Entity,
     clip: Option<Rect>,
 ) {
-    let (node, global_transform, style, calculated_clip) = node_query.get_mut(entity).unwrap();
-    // Update this node's CalculatedClip component
-    match (clip, calculated_clip) {
+    let (node, global_transform, style, old_clip) = node_query.get_mut(entity).unwrap();
```

Comment:
I think `maybe_calculated_clip` is better than `old_clip`, makes it clearer it's an optional smart pointer to the component.

```suggestion
    let (node, global_transform, style, maybe_calculated_clip) = node_query.get_mut(entity).unwrap();
```